101-custom-rp-with-function/README.md

```diff
@@ -7,89 +7,67 @@
     <img src="http://armviz.io/visualizebutton.png"/>
 </a>
 
-This sample template deploys a custom resource provider to Azure and creates a user using an ARM template.  You may optionally deploy the function app and custom provider definition - this only needs to be done once and then the custom resource can be deployed many times.
+This sample Azure Resource Manager template deploys a custom resource provider to Azure that extends the Azure Resource Manager API set with custom abilities and then utilizes the new extension to deploy "customResources".
 
-The customproviders resource is a hidden Azure resource so to confirm that the resource provider has been deployed you will have to check the box that says *Show hidden types* in the Azure portal Overview page for the resource group.
+The custom resource provider is a hidden Azure resource so to confirm that the custom resource provider has been deployed you will have to check the box that says *Show hidden types* in the Azure portal Overview page for the resource group.
 
 ![](images/showhidden.png)
 
 ## Details on the custom resource provider created
 
-This sample deployment creates the following two apis on the resource.
+This sample deployment creates the following apis on the resource.
 
-1) An ARM extended resource called "users"
-2) An API called "ping"
+1) An Azure Resource Manager extended API called "customResources"
 
-### Users
+### CustomResources
 
-The users resource is defined in the following part of the ARM template : 
+"customResources" is an API extension contained within the custom resource provider. This extension is defined in the following part of the ARM template under the "resourceProviders" resource:
 
 ```json
 "resourceTypes": [
-                                {
-                                    "name":"users",
-                                    "routingType":"Proxy,Cache",
-                                    "endpoint": "[concat('https://', parameters('funcname'), '.azurewebsites.net/api/{requestPath}')]"
-                                }
-                            ]
+                    {
+                        "name": "customResources",
```

Comment:
Let's change this name - we're using the type as the name which is a bit confusing... maybe try something like "myCustomResource" to show that this is an instance of the type.

Reply:
This is still the type name. The instance name is entered through the parameters file.

Reply:
is it a singleton?

Reply:
nah, this is the type that is created in ARM. Similar to how virtualMachines is a type then you can create instances off of it. The name field is the type name customResources is the type. When you create a resource off the custom provider it would be created off /subscriptions/sub/resourceGroups/rg/providers/Microsoft.CustomProviders/resourceProviders/myRpName/customResources/myCustomResourceName